homeassistant/components/frigidaire/config_flow.py

```diff
@@ -0,0 +1,84 @@
+"""Config flow for frigidaire integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import frigidaire
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema({"username": str, "password": str})
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    def setup(username: str, password: str) -> list[frigidaire.Appliance]:
+        try:
+            client = frigidaire.Frigidaire(username, password)
+            return client.get_appliances()
+        except frigidaire.FrigidaireException as err:
```

Comment:
Looking at the upstream code, this can also be throw if we can't connect... 
Can we split those cases?